src/utils/preview.ts

```diff
@@ -0,0 +1,81 @@
+import { createContext } from 'react';
+import { getPageContent, HeadingNode, PageContent } from './content';
+import { Environment } from './env';
+import { IRenderError, RenderError } from './error';
+import { mdxSerialize } from './mdx-serialize';
+import { Properties, SlugProperties } from './properties';
+
+export type PreviewMode = boolean;
+
+export const PreviewModeContext = createContext<PreviewMode>(false);
+
+export type PreviewPageProps = {
+  env: Environment;
+  properties: SlugProperties;
+  headings: HeadingNode[];
+  source?: string;
+  content?: PageContent;
+  error?: IRenderError;
+};
+
+export async function buildPreviewProps({
```

Comment:
this was trying to replicate the getStaticProps of the non-preview mode as much as i could, maybe we want to refactor this into hooks though